tests/unit/h/search/query_test.py

```diff
@@ -691,27 +641,137 @@ def search(self, search):
         return search
 
 
-@pytest.mark.usefixtures("pyramid_config")
-class TestHiddenFilter:
+class TestGroupAndModerationFilter:
+    def test_matches_only_annotations_from_specified_groups(
+        self, search, Annotation, groups, group_service, pyramid_request
+    ):
+        group_service.groups_readable_by.return_value = groups
+        Annotation(groupid="other_group")
+        annotation_ids = [Annotation(groupid=group.pubid).id for group in groups]
+
+        result = search.run(MultiDict(("group", group.pubid) for group in groups))
+
+        group_service.groups_readable_by.assert_called_with(
+            pyramid_request.user, group_ids=[group.pubid for group in groups]
+        )
+        assert sorted(result.annotation_ids) == sorted(annotation_ids)
+
+    def test_matches_only_annotations_in_groups_readable_by_user(
+        self, search, Annotation, group_service, factories
+    ):
+        readable_group = factories.Group(pubid="readable_group")
+        group_service.groups_readable_by.return_value = [readable_group]
+        Annotation(groupid="unreadable_group", shared=True)
+        annotation_ids = [
+            Annotation(groupid="readable_group").id,
+            Annotation(groupid="readable_group").id,
+        ]
+
+        result = search.run(MultiDict({}))
+
+        assert sorted(result.annotation_ids) == sorted(annotation_ids)
+
     @pytest.mark.usefixtures("as_user")
     def test_visibility_annotations_by_others(
         self, search, make_annotation, banned_user, is_nipsaed, is_hidden
     ):
         annotation = make_annotation(banned_user)
 
-        result = search.run({})
+        result = search.run(MultiDict({}))
 
         if is_nipsaed or is_hidden:
             # We should not see the annotation
             assert not result.annotation_ids
         else:
             assert result.annotation_ids == [annotation.id]
 
+    @pytest.mark.usefixtures("as_user")
+    def test_moderators_see_hidden_annotations(
+        self,
+        search,
+        factories,
+        user,
+        group_service,
+        identity_permits,
+        make_annotation,
```

Comment:
I don't particularly like this type of fixture.

This is just a function that is itself parameterized based on on other fixtures.

So whether or not you use is_nipsaed and is_hidden that function is going to be called 4 times.

I'd much prefer to just have regular parameters in that function that I can chose to call with fixture values if I need to.

I prefer to have the rest of the code reviewed and then focus on a test refactor if needed.